src/barril/units/unit_database.py

```diff
@@ -478,8 +496,11 @@ def AddCategory(
         if valid_units is not None:
             # valid units given: check if all the given units are valid
             quantity_units = set(self.GetUnits(quantity_type))
-            for unit in valid_units:
-                if unit not in quantity_units:
+            for i, unit in enumerate(valid_units):
+                was_unit_fixed, fixed_unit = FixUnitIfIsLegacy(unit)
+                if was_unit_fixed:
+                    valid_units[i] = fixed_unit
+                if fixed_unit not in quantity_units:
```

Comment:
Question.

Do we add units in the legacy format?
Originally this fix" will affect only units loaded from old files turning then to the new format.